src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectDefaultConverter.cs

```diff
@@ -16,6 +16,9 @@ internal class ObjectDefaultConverter<T> : JsonObjectConverter<T> where T : notn
     {
         internal override bool CanHaveIdMetadata => true;
 
+#if NET6_0_OR_GREATER
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
```

Comment:
Out of curiosity, what are some of the optimizations that were being missed without this attribute? I had tried using it a few months back on the `JsonConverter<T>.TryWrite` method (which is also fairly large) and did not receive a substantial perf boost.

Reply:
I am not sure how to find this out, with tiered jitting and all (e.g. hit a breakpoint after running a 1,000 times to inspect the native code?). Also, I did not run with `crossgen2` yet, but typically that has the same results once the warm-up phase is done.

The results are small when there are a couple properties but should be greater when there are more properties. I had to run benchmarks a few times to make sure noise isn't a factor. I welcome investigation from others on this.

Here's some results with and without `AggressiveOptimization`. Again, I assume the more properties, the more savings. Also I tend to focus on the `Min` column more than `Mean`.

<details>
  <summary>With AO and without</summary>

Using a class with 2 string properties; I ran 5 times and took the fastest.

With AO:
|      Method |     Mean |   Error |  StdDev |  Gen 0 | Allocated |
|------------ |---------:|--------:|--------:|-------:|----------:|
|   Serialize | 197.1 ns | 0.34 ns | 0.27 ns |      - |         - |
| Deserialize | 330.1 ns | 0.42 ns | 0.38 ns | 0.0161 |     104 B |

Without AO:
|      Method |     Mean |   Error |  StdDev |  Gen 0 | Allocated |
|------------ |---------:|--------:|--------:|-------:|----------:|
|   Serialize | 198.8 ns | 0.10 ns | 0.09 ns |      - |         - |
| Deserialize | 333.1 ns | 1.14 ns | 1.01 ns | 0.0161 |     104 B |

Running `ReadJson<LargeStructWithProperties>` and `WriteJson<LargeStructWithProperties>` since it has 10 properties. I ran the benchmarks 3 times and took the fastest. The "Stream" cases didn't get any faster -- not sure why, although that does have a different code path in the object converter.

With AO:
|                   Method |     Mean |    Error |   StdDev |   Median |      Min |      Max |  Gen 0 | Allocated |
|------------------------ |---------:|---------:|---------:|---------:|---------:|---------:|-------:|----------:|
|       SerializeToString | 650.1 ns |  5.49 ns |  4.87 ns | 648.0 ns | 643.8 ns | 656.6 ns | 0.0765 |     488 B |
|    SerializeToUtf8Bytes | 618.7 ns |  7.81 ns |  7.31 ns | 615.9 ns | 609.7 ns | 633.3 ns | 0.0577 |     376 B |
|       SerializeToStream | 766.6 ns | 10.86 ns |  9.63 ns | 766.1 ns | 754.1 ns | 783.0 ns | 0.0368 |     232 B |
| SerializeObjectProperty | 831.6 ns | 13.94 ns | 11.64 ns | 828.7 ns | 819.1 ns | 862.5 ns | 0.1347 |     848 B |

Without AO:
|                  Method |     Mean |   Error |  StdDev |   Median |      Min |      Max |  Gen 0 | Allocated |
|------------------------ |---------:|--------:|--------:|---------:|---------:|---------:|-------:|----------:|
|       SerializeToString | 667.1 ns | 2.42 ns | 2.15 ns | 667.5 ns | 661.5 ns | 669.4 ns | 0.0754 |     488 B |
|    SerializeToUtf8Bytes | 620.1 ns | 3.00 ns | 2.66 ns | 620.0 ns | 615.0 ns | 624.4 ns | 0.0598 |     376 B |
|       SerializeToStream | 760.7 ns | 0.80 ns | 0.67 ns | 760.6 ns | 759.8 ns | 762.1 ns | 0.0370 |     232 B |
| SerializeObjectProperty | 839.5 ns | 8.38 ns | 7.84 ns | 838.3 ns | 826.0 ns | 851.7 ns | 0.1340 |     848 B |

With AO:
|                   Method |     Mean |     Error |    StdDev |   Median |      Min |      Max |  Gen 0 | Allocated |
|------------------------- |---------:|----------:|----------:|---------:|---------:|---------:|-------:|----------:|
|    DeserializeFromString | 1.114 us | 0.0083 us | 0.0070 us | 1.114 us | 1.104 us | 1.131 us | 0.0318 |     200 B |
| DeserializeFromUtf8Bytes | 1.042 us | 0.0118 us | 0.0111 us | 1.040 us | 1.028 us | 1.065 us | 0.0289 |     200 B |
|    DeserializeFromStream | 1.519 us | 0.0189 us | 0.0176 us | 1.516 us | 1.493 us | 1.556 us | 0.0478 |     328 B |

Without AO:
|                   Method |     Mean |     Error |    StdDev |   Median |      Min |      Max |  Gen 0 | Allocated |
|------------------------- |---------:|----------:|----------:|---------:|---------:|---------:|-------:|----------:|
|    DeserializeFromString | 1.151 us | 0.0046 us | 0.0041 us | 1.152 us | 1.143 us | 1.156 us | 0.0276 |     200 B |
| DeserializeFromUtf8Bytes | 1.063 us | 0.0141 us | 0.0132 us | 1.065 us | 1.043 us | 1.081 us | 0.0295 |     200 B |
|    DeserializeFromStream | 1.497 us | 0.0053 us | 0.0047 us | 1.496 us | 1.489 us | 1.507 us | 0.0479 |     328 B |

</details>

Reply:
In the end, after running thousands of times, the jitted code may be more or less the same with or without the [AO]. But [AO] should produce faster code sooner.

Reply:
Can we please avoid [AO]?
Methods with [AO] aren't instrumented so won't be able to benefit from PGO (devirtualize calls, move cold parts of methods further from hot ones, etc) - e.g. run your benchmark again with `DOTNET_ReadyToRun=0`, `DOTNET_TC_QuickJitForLoops=1`, `DOTNET_TieredPGO=1` with and without `[AO]`

Also, when a method is promoted to tier1 naturally (without [AO]`) we can:
1) Get rid of static initialization if we promote a method naturally from tier0 -> tier1.
2) Convert `static readonly` fields to constants 
3) Inliner is able to resolve `call` tokens and produce better results

/cc @AndyAyersMS

Reply:
[AO] should be avoided. It causes methods to be optimized prematurely. We use it only in a handful of places where we feel like we can't afford to ever run unoptimized code.

Reply:
Thanks, I'll re-run with those settings.